lora/README.md

```diff
@@ -0,0 +1,113 @@
+MF_LORA_ADAPTER_LOG_LEVEL=[Lora Adapter Log Level] MF_NATS_URL=[NATS instance URL] MF_LORA_ADAPTER_LORA_MESSAGE_URL=[LoRa Server mqtt broker URL] MF_LORA_ADAPTER_ROUTEMAP_URL=[Lora adapter routemap URL] MF_LORA_ADAPTER_ROUTEMAP_PASS=[Lora adapter routemap password] MF_LORA_ADAPTER_ROUTEMAP_DB=[Lora adapter routemap instance] MF_THINGS_ES_URL=[Things service event store URL] MF_THINGS_ES_PASS=[Things service event store password] MF_THINGS_ES_DB=[Things service event store db] MF_LORA_ADAPTER_INSTANCE_NAME=[LoRa adapter instance name] $GOBIN/mainflux-lora
+```
+
+### Using docker-compose
+
+This service can be deployed using docker containers.
+Docker compose file is available in `<project_root>/docker/addons/lora-adapter/docker-compose.yml`. In order to run Mainflux lora-adapter, execute the following command:
+
+```bash
+docker-compose -f docker/addons/lora-adapter/docker-compose.yml up -d
+```
+
+## Usage
+
+First of all you must run your LoRa Server. You need to clone the repository with command `go get github.com/brocaar/loraserver-docker` and deploy docker containers with command `docker-compose up` from `$GOPATH/src/github.com/brocaar/loraserver-docker` repository.
+Then you must provision your system. Basically it means create a Network-server where to connect your Gateways and Applications where to connect Devices. If you need more information about the LoRa Server setup you can check [the LoRa App Server documentation](https://www.loraserver.io/lora-app-server/overview/)
+
+Once you are done with the LoRa Server setup you can run the lora-adapter. This service uses RedisDB to create a route map between both systems. As in Mainflux we use Channels to connect Things, Loraserser uses Applications to connect Devices. Route map connects applicationsID with channelID and deviceEUI with thingID.
+The lora-adapter uses the matadata of provision events emitted by mainflux-things service to update the route map.
+For that, you must provision Mainflux Channels and Things with an extra `metadata` key in the JSON Body of the HTTP request. It must be a JSON object with keys `type` and `appID` or `devEUI`. Obviously `type` must be `lora`, `appID` and `devEUI` must be an existent Lora application ID and device EUI.
+
+```
+{
+  "name": "<channel name>",
+  "metadata:":{
+    "type": "lora",
+    "appID": "<application ID>"
+  }
+}
+
+```
+```
+{
+  "type": "device",
+  "name": "<thing name>",
+  "metadata:":{
+    "type": "lora",
+    "devEUI": "<device EUI>"
+  }
+}
+```
+
+To receive Lora messages the lora-adapter subscribes to topic `applications/+/devices/+` of the LoRa Server MQTT broker. The [LoRa Gateway Bridge](https://www.loraserver.io/lora-gateway-bridge/overview/)uses the same topic to publish decoded messages received from gateways as UDP packets. The lora-adapter verify the applicationID and the deviceEUI of published message and if they are known it forwards the message on the Mainflux NATS broker as corresponding channel and thing.
```

Comment:
I  assume that `+`  are `<lora_application_id>` and `<device_eui>`, respectively. Maybe it would be clearer if you replace `+` with what it actually represents.

Reply:
@dusanb94 no, `+` is a joker (wildcard) topic of MQTT - can be anything - https://www.hivemq.com/blog/mqtt-essentials-part-5-mqtt-topics-best-practices

Reply:
No it isn't. This topic means that you will subscribe to all messages from all apps and all devices.